api/adaptive_load/benchmark_result.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+
+package nighthawk.adaptive_load;
+
+import "api/adaptive_load/metric_spec.proto";
+import "api/client/output.proto";
+
+// Binary result of checking a metric value against a threshold.
+enum SimpleThresholdStatus {
+  // Value should only occur when the field is unset.
+  UNKNOWN_THRESHOLD_STATUS = 0;
+  // The metric was below an upper threshold or above a lower threshold.
+  WITHIN_THRESHOLD = 1;
```

Comment:
Why are we choosing to do both the binary result and the threshold score? I thought we were going towards the score only.

Reply:
I'm now leaning toward a more general approach from the start, where we get rid of the binary result entirely, even for basic cases. It would simplify the code quite a bit by removing branches.

ExponentialSearchStepController can look at the sign of the score and ignore the magnitude.

To avoid confusion, we can provide StepFunctionScoringFunction that returns 1.0 for within threshold, -1.0 otherwise, for use with ExponentialSearchStepController. This would be better than making the user choose LinearScoringFunction with a dummy k that has no effect.